src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -165,11 +165,7 @@ public HaplotypeCallerEngine(final HaplotypeCallerArgumentCollection hcArgs, Ass
         forceCallingAllelesPresent = hcArgs.alleles != null;
         initialize(createBamOutIndex, createBamOutMD5);
         if (hcArgs.assemblyStateOutput != null) {
-            try {
-                assemblyDebugOutStream = new PrintStream(Files.newOutputStream(IOUtils.getPath(hcArgs.assemblyStateOutput)));
-            } catch (IOException e) {
-                throw new UserException.CouldNotCreateOutputFile(hcArgs.assemblyStateOutput, "Provided argument for assembly debug graph location could not be created");
-            }
+            assemblyDebugOutStream = new PrintStream(hcArgs.assemblyStateOutput.getOutputStream());
```

Comment:
Here too we need to handle errors during stream creation. If `IOException` is no longer thrown then I guess we have to call `checkError()`? (unless you can find a better option -- it would be preferable to be able to catch `IOException` like before)

Reply:
For this one, I'll change `assemblyDebugOutStream` to be an OutputStreamWriter, with appropriate catch clauses where its used.